docs/reference/esql/esql-syntax.asciidoc

```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
 ----
 ====
 
+[discrete]
+[[esql-identifiers]]
+==== Identifiers
+
+Unless containing special characters, the identifiers can be used as-are and
+don't require quoting. If they do, however, they must be quoted with
+backticks (```). The special characters are those that have already a semantic
+meaning for a given command, like arithmetic operators (`-`, `+` etc.),
+decimals delimiters (`.`), field delimiters (`,`), command separators (`|`),
+subfield referencing (`.`) a.s.o. This is also applicable to identifiers that
+could be interpreted like numbers (the scientific notation, like `1e2`) or
+simply start with a number.
+
+For instance:
+
+[source,esql]
+----
+// Retain just one field
+FROM index
+| KEEP 1.field
+----
+
+is legal. However, if same field is to be used with an <<esql-eval, EVAL>>,
+it'd have to be quoted:
+
+[source,esql]
+----
+// Copy one field
+FROM index
+| EVAL my_field = `1.field`
+----
+
+[discrete]
+[[esql-literals]]
+==== Literals
+
+{esql} supports currently numeric and string literals. The numeric literals
+can be used as are. The string literals need to be quoted in double
```

Comment:
`as they are` or `as is`. I believe `as are` is incorrect.

Reply:
Reworked this paragraph too, splitting strings and numeric literals.